cms/djangoapps/contentstore/helpers.py

```diff
@@ -316,9 +316,9 @@ def _import_xml_node_to_parent(
     block_type = node.tag
 
     # Generate the new ID:
-    id_generator = ImportIdGenerator(parent_key.context_key)
-    def_id = id_generator.create_definition(block_type, slug_hint)
-    usage_id = id_generator.create_usage(def_id)
+    runtime.id_generator = ImportIdGenerator(parent_key.context_key)
```

Comment:
For this specific case runtime was using [SplitMongoIdManager](https://github.com/openedx/edx-platform/blob/acc86dd3fdf8fe7840f7f4ece99cf980d3c72d2f/xmodule/modulestore/split_mongo/id_manager.py#L14) instead of ImportIdGenerator. 

Details: [Unit test failure](https://github.com/openedx/edx-platform/actions/runs/8507257288/job/23298811363#logs) : [Raw Logs](https://pipelinesghubeus2.actions.githubusercontent.com/Dxro1FLYwwaYoTvxXxrOP6MlKDWB8jnvCh9RqeQWY1bZ0SbUcd/_apis/pipelines/1/runs/389910/signedlogcontent/14?urlExpires=2024-04-02T09%3A26%3A41.1706040Z&urlSigningMethod=HMACV1&urlSignature=u6qIYikKjB5c066TLK3uGMUgT7TPZ3IqsKbRa2vK8cg%3D)

In general `id_generator` should be assigned by `runtime` but for every case where a custom `id_generator` was required we have to pass it through attribute. i.e. `runtime.id_generator`

Reply:
Ohhhh, that is interesting. I guess it makes sense--the runtime here is the general LMS/CMS runtime (CachingDescriptorSystem), so by default, its id_generator is not going to match ImportSystem's id_generator. Thanks for pointing that out @irtazaakram .

Setting it here is a little awkward, but I don't know of a better solution currently.

In order to contain the impact of this, do you think it would be reasonable to set the original id_generator to a temporary value, and then restore it? If you agree, could you make that change, and add a comment explaining why we need to do it?
```python
original_id_generator = runtime.id_generator
runtime.id_generator = ImportIdGenerator(...)
...
runtime.id_generator = original_id_generator
```

Reply:
done. https://github.com/openedx/edx-platform/pull/34398/commits/682a80adddfca953f86b8fb7464f06b6e9d34357